src/RedisResourceManager.php

```diff
@@ -22,17 +22,33 @@
 use function method_exists;
 use function parse_url;
 use function str_replace;
+use function str_starts_with;
 use function strpos;
 use function strtoupper;
 use function trim;
 
 /**
  * This is a resource manager for redis
  */
+/**
+ * @psalm-type ResourceArrayShape =array{
+ *     resource?: RedisResource,
+ *     database?: non-empty-string,
+ *     lib_options?: array,
+ *     version?: int|string,
+ *     password?: non-empty-string,
+ *     persistent_id?: non-empty-string,
+ *     server?: non-empty-string,
```

Comment:
`server` is actually an array which is getting normalized to:

```php
$server = [
    'host'    => $host,
    'port'    => $port,
    'timeout' => $timeout,
];
```

within `normalizeServer`. 

```suggestion
 *     server: array{host: string, port: int, timeout: int},
```